src/bootloaders/extlinux.c

```diff
@@ -35,33 +35,44 @@
 static KernelArray *kernel_queue = NULL;
 static char *extlinux_cmd = NULL;
 static char *base_path = NULL;
+static char *sgdisk_cmd = NULL;
 
 static bool extlinux_init(const BootManager *manager)
 {
         autofree(char) *ldlinux = NULL;
         const char *prefix = NULL;
         autofree(char) *boot_device = NULL;
+        autofree(char) *parent_disk = NULL;
+        int partition_index;
 
         if (kernel_queue) {
                 kernel_array_free(kernel_queue);
         }
+
         kernel_queue = nc_array_new();
         if (!kernel_queue) {
                 DECLARE_OOM();
                 abort();
         }
+
         if (base_path) {
                 free(base_path);
                 base_path = NULL;
         }
-        base_path = boot_manager_get_boot_dir((BootManager *)manager);
-        OOM_CHECK_RET(base_path, false);
 
         if (extlinux_cmd) {
                 free(extlinux_cmd);
                 extlinux_cmd = NULL;
         }
 
+        if (sgdisk_cmd) {
+                free(sgdisk_cmd);
+                sgdisk_cmd = NULL;
+        }
+
+        base_path = boot_manager_get_boot_dir((BootManager *)manager);
+        OOM_CHECK_RET(base_path, false);
```

Comment:
Why was this moved?

Reply:
just a logic organization, keeping all these global checks up in the beginning - no more than stylistic organization.